flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SystemProcessingTimeService.java

```diff
@@ -199,12 +214,20 @@ int getNumTasksScheduled() {
 	 */
 	private static final class TriggerTask implements Runnable {
 
+		private final SystemProcessingTimeService timerService;
 		private final Object lock;
 		private final ProcessingTimeCallback target;
 		private final long timestamp;
 		private final AsyncExceptionHandler exceptionHandler;
 
-		TriggerTask(AsyncExceptionHandler exceptionHandler, final Object lock, ProcessingTimeCallback target, long timestamp) {
+		private TriggerTask(
```

Comment:
We should only hand in the `status` field here, not the complete timer service because that potentially exposes to many things.